tutorial/guias/Paso_a_paso_DIA_4.md

```diff
@@ -0,0 +1,204 @@
+
+Finalmente la última línea nos da un resumen de cuántos tests pasaron, cuántos fallaron y qué tiempo tomó ejecutarlos todos. En nuestro caso, nuestro único test fue exitoso :smiley:.
+
+A modo de ejemplo, agreguemos a nuestra clase un test que sabemos que va a fallar:
+
+```python
+from unittest import TestCase
+
+from src.model.fruit import Fruit
+
+
+class TestFruit(TestCase):
+    def setUp(self):
+        self.fruit = Fruit((10, 10))
+
+    def test_created_as_dataclass(self):
+        self.assertEqual(self.fruit.position, (10, 10))
+    
+    def test_fail(self):
+        self.assertEqual(1, 2)
+
+```
+
+Si volvemos a ejecutar el comando `pytest`, obtenemos la siguiente salida:
+
+```bash
+========================= test session starts ==========================================
+platform linux -- Python 3.10.6, pytest-7.2.2, pluggy-1.0.0
+rootdir: /media/jj/2C3E97D93E979A82/Storage/personal-projects/codexl/snake/tutorial/DIA 4/snake
+plugins: web3-5.30.0, anyio-3.6.1
+collected 2 items                                                                                                                        
+
+tests/test_fruit.py .F                 [100%]
+
+========================== FAILURES =====================================================
+______________________ TestFruit.test_fail ______________________________________________
+
+self = <tests.test_fruit.TestFruit testMethod=test_fail>
+
+    def test_fail(self):
+>       self.assertEqual(1, 2)
+E       AssertionError: 1 != 2
+
+tests/test_fruit.py:14: AssertionError
+========================== short test summary info =======================================
+FAILED tests/test_fruit.py::TestFruit::test_fail - AssertionError: 1 != 2
+========================== 1 failed, 1 passed in 0.28s ===================================
+```
+
+A partir de la línea que comienza con *collected* vemos que ahora se detectaron dos tests. En la siguiente línea tenemos el archivo que contiene los tests seguido de un punto y una `F`. Ya sabíamos que íbamos a tener un fallo porque nuestro último test verifica que 1 sea igual a 2. Las siguientes líneas nos dan detalles de este fallo, en qué parte de nuestro código ocurre el error y qué tipo de error es. Luego tenemos un resumen corto del fallo y finalmente tenemos 1 test exitoso, 1 fallido y todo se ejecutó en 0.28 segundos. Antes de continuar, asegúrate de borrar la función `test_fail` para no volver a tener tests fallidos.
+
+### Clase `Snake`
+
+Ahora agreguemos las pruebas para la clase `Snake`:
+
+Archivo **tests/test_snake.py**:
+
+```python
+from unittest import TestCase
+
+from src.model.snake import Direction, Snake
+
+
+class TestSnake(TestCase):
+    def setUp(self):
+        self.snake = Snake(position=(50, 50), direction=Direction.RIGHT, size=4)
+
+    def test_create_body(self):
```

Comment:
Should be add a note saying that usually these tests should start with the prefix `test`. I think that can be modified to either start with `test` or `test_`, or even end with `test` or `_test`. But the default I think it is starting with `test`.

Reply:
I say something similar above when talking about snake_casing. I don't know if adding a note would be too repetitive.